src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -220,7 +220,7 @@ public partial class MarkStep : IStep
 		public MarkStep ()
 		{
 			_methods = new Queue<(MethodDefinition, DependencyInfo, MessageOrigin)> ();
-			_virtual_methods = new HashSet<(MethodDefinition, MarkScopeStack.Scope)> ();
+			_virtual_methods = new Dictionary<MethodDefinition, MarkScopeStack.Scope> ();
```

Comment:
Is the perf improvement due to the change of key? If so is `MethodDefinition` actually a good option for lookups? It doesn't implement `IEquatable` and doesn't have a custom `GetHashcode`. Would a custom record using what makes a `MethodDefinition` identity better then?

Reply:
The trimmer only creates a single `MethodDefinition` object for each method definition in metadata, so `object`'s implementation of `GetHashCode` and `Equals` are fine here.